src/Driver/SolverTypes/IMEXSolverType.jl

```diff
@@ -0,0 +1,160 @@
+
+export IMEXSolverType
+
+"""
+# Description
+    IMEXSolverType(;
+        splitting_type = HEVISplitting(),
+        linear_model = AtmosAcousticGravityLinearModel,
+        implicit_solver = ManyColumnLU,
+        implicit_solver_adjustable = false,
+        solver_method = ARK2GiraldoKellyConstantinescu,
+        solver_storage_variant = LowStorageVariant(),
+    )
+
+This solver type constructs a solver for ODEs with the
+additively-partitioned form:
+
+```math
+  \\dot{Q} = [l(Q, t)] + [f(Q, t) - l(Q, t)]
+```
+
+where `Q` is the state, `f` is the full tendency and `l` is the chosen
+implicit operator.
+
+# Arguments
+- `splitting_type` (DiscreteSplittingType): The type of discrete
+    splitting to apply to the right-hand side.
+    Default: `SlowFastSplitting()`
+- `linear_model` (Type): The linear model describing fast dynamics.
+    Default: `AtmosAcousticGravityLinearModel`
+- `implicit_solver` (Type): A linear solver for inverting the
+    implicit system of equations.
+    Default: `ManyColumnLU`
+- `implicit_solver_adjustable` (Bool): A flag identifying whether
+    or not the `implicit_solver` can be updated as the time-step
+    size changes.
+    Default: `false`
+- `solver_method` (Function): Function defining the particular additive
+    Runge-Kutta method to be used for the IMEX method.
+    Default: `ARK2GiraldoKellyConstantinescu`
+- `solver_storage_variant` (Type): Storage type for the additive
+    Runge-Kutta method.
+    Default: `LowStorageVariant()`
+
+### References
+    @article{giraldo2013implicit,
+      title={Implicit-explicit formulations of a three-dimensional
+             nonhydrostatic unified model of the atmosphere ({NUMA})},
+      author={Giraldo, Francis X and Kelly, James F and Constantinescu, Emil M},
+      journal={SIAM Journal on Scientific Computing},
+      volume={35},
+      number={5},
+      pages={B1162--B1194},
+      year={2013},
+      publisher={SIAM}
+    }
+"""
+struct IMEXSolverType{DS, ST} <: AbstractSolverType
+    # The type of discrete splitting to apply to the right-hand side
+    splitting_type::DS
+    # Linear model describing fast dynamics
+    linear_model::Type
```

Comment:
Maybe `implicit_model` would be better to be consistent with `implicit_solver` and to cover future generalizations.

Reply:
Just a reminder -- if you change this, you'll have to change some tests, experiments and tutorials that are constructing `IMEXSolverType`s.